pkg/backend/filestate/restblob/restblob.go

```diff
@@ -0,0 +1,418 @@
+package restblob
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gocloud.dev/blob"
+	"gocloud.dev/blob/driver"
+	"gocloud.dev/gcerrors"
+)
+
+func init() {
+	blob.DefaultURLMux().RegisterBucket("resthttp", &RestBlob{})
```

Comment:
maybe it would be better of calling it `tfremote`, `tfstore` (with respecting `tfremotes` or `tfstores`), as it's not just any rest backend, but following the way terraform expects remote stores to work?